sklearn/tree/_tree.pyx

```diff
@@ -391,29 +548,49 @@ cdef class Entropy(ClassificationCriterion):
 
         cdef double entropy = 0.0
         cdef double total = 0.0
-        cdef double tmp
+        cdef double pmk
```

Comment:
I think `probability_mass_k` is more explicit. The statements are short so we can afford a long variable name here.